File: parsing/arraysum/arraysum.c
/*
Array Sum
Autor:
    ?
Colaborador:
    Leonardo da Mata barroca at gmail.com
Descricao:
    Esse programa recebe como parametro um nome 
    de um arquivo na entrada, abre esse arquivo 
    para leitura, le os números presentes no 
    arquivo (um numero por linha) e realiza a
    soma desses números.
Complexidade:
    ?
Dificuldade:
    facil
Licenca:
    GPL
 */

#include <stdio.h>
#include <stdlib.h>
int main(int argc, char * argv[]){
    if (argc != 2){
        printf("Usage: %s <file_name>\n",argv[0]);
        exit(1);
    }
    int array_size = 0;
    int* array;
    int i=0;
    int sum=0;
    int num=0;
    FILE *fp; 
    array = (int*)malloc(sizeof(int)*1024);
    fp=fopen(argv[1],"r");
    if(fp==NULL){
        printf("Error opening file\n");
        exit(1);
    }
    while(fscanf(fp, "%d", &num) != EOF){
        array[i]=num;
        i++;
    }
    array_size=i;
    for(i=0;i<array_size;i++){
        sum+=array[i];
    }
    printf("Result: %d\n",sum);
    return 0;
}

File: algorithms/parsing/arraysum.py
# coding: utf-8
'''
Array Sum
Autor:
    ?
Colaborador:
    Dayvid Victor (victor.dvro@gmail.com)
Descricao:
    Esse programa recebe como parametro uma lista
    e retorna a soma dos elementos desta lista.
Complexidade:
    O(n)
Dificuldade:
    facil
Licenca:
    GPL
'''

def arraysum(l, key = lambda a, b: a + b):
	s = 0
	for e in l:
		s = key(s,e)
	return s

if __name__ == '__main__':
	l1 = [1,2,3,4,5,6,7,8,9,10]
	l2 = [-4,-3,-2,-1,0,1,2,3,4]

	print arraysum(l1)
	print arraysum(l2)

